REGRESSION(269235@main?):[ Monterey+ ] http/tests/security/xss-DENIED-script-inject-into-inactive-window2-pson.html is flaky recently

sideshowbarker opened the pull request:
#### c586c6d289d7c653602ddadbdb6ccab49671ba34
<pre>
REGRESSION(269235@main?):[ Monterey+ ] http/tests/security/xss-DENIED-script-inject-into-inactive-window2-pson.html is flaky recently
<a href="https://bugs.webkit.org/show_bug.cgi?id=263879">https://bugs.webkit.org/show_bug.cgi?id=263879</a>

Reviewed by Chris Dumez.

This change updates the
http/tests/security/xss-DENIED-script-inject-into-inactive-window2-pson.html
test to ensure it consistently passes.

The change adds a new isFullyActive(Document&amp;) function to window.internals,
and uses that function in the test.

* LayoutTests/TestExpectations:
* LayoutTests/http/tests/security/xss-DENIED-script-inject-into-inactive-window2-pson-expected.txt:
* LayoutTests/http/tests/security/xss-DENIED-script-inject-into-inactive-window2-pson.html:
* Source/WebCore/dom/Document.h:
* Source/WebCore/testing/Internals.cpp:
(WebCore::Internals::isFullyActive):
* Source/WebCore/testing/Internals.h:
* Source/WebCore/testing/Internals.idl:

Canonical link: <a href="https://commits.webkit.org/270737@main">https://commits.webkit.org/270737@main</a>
</pre>
<!--EWS-Status-Bubble-Start-->
https://github.com/WebKit/WebKit/commit/91ca8f47780108a9bebadff3ecf98bd3702224df

| Misc | iOS, tvOS & watchOS  | macOS  | Linux |  Windows |
| ----- | ---------------------- | ------- |  ----- |  --------- |
| [✅ 🧪 style](https://ews-build.webkit.org/#/builders/38/builds/26054 "Passed style check") | [✅ 🛠 ios](https://ews-build.webkit.org/#/builders/48/builds/4665 "Built successfully") | [✅ 🛠 mac](https://ews-build.webkit.org/#/builders/14/builds/27331 "Built successfully") | [✅ 🛠 wpe](https://ews-build.webkit.org/#/builders/5/builds/28152 "Built successfully") | [✅ 🛠 wincairo](https://ews-build.webkit.org/#/builders/32/builds/23857 "Built successfully") 
| [✅ 🧪 bindings](https://ews-build.webkit.org/#/builders/9/builds/26376 "Passed tests") | [✅ 🛠 ios-sim](https://ews-build.webkit.org/#/builders/49/builds/6429 "Built successfully") | [✅ 🛠 mac-AS-debug](https://ews-build.webkit.org/#/builders/51/builds/2087 "Built successfully") | [✅ 🧪 wpe-wk2](https://ews-build.webkit.org/#/builders/34/builds/23920 "Passed tests") | 
| [✅ 🧪 webkitperl](https://ews-build.webkit.org/#/builders/11/builds/26308 "Passed tests") | [✅ 🧪 ios-wk2](https://ews-build.webkit.org/#/builders/47/builds/3534 "Passed tests") | [✅ 🧪 api-mac](https://ews-build.webkit.org/#/builders/18/builds/22448 "Passed tests") | [✅ 🛠 gtk](https://ews-build.webkit.org/#/builders/2/builds/28728 "Built successfully") | 
| | [✅ 🧪 ios-wk2-wpt](https://ews-build.webkit.org/#/builders/42/builds/3153 "Passed tests") | [✅ 🧪 mac-wk1](https://ews-build.webkit.org/#/builders/10/builds/23400 "Passed tests") | [⏳ 🧪 gtk-wk2](https://ews-build.webkit.org/#/builders/GTK-WK2-Tests-EWS "Waiting to run tests") | 
| | [✅ 🧪 api-ios](https://ews-build.webkit.org/#/builders/13/builds/23770 "Passed tests") | [✅ 🧪 mac-wk2](https://ews-build.webkit.org/#/builders/36/builds/23780 "Passed tests") | [⏳ 🧪 api-gtk](https://ews-build.webkit.org/#/builders/API-Tests-GTK-EWS "Waiting to run tests") | 
| | [✅ 🛠 tv](https://ews-build.webkit.org/#/builders/44/builds/3187 "Built successfully") | [✅ 🧪 mac-AS-debug-wk2](https://ews-build.webkit.org/#/builders/50/builds/1390 "Passed tests") | | 
| | [✅ 🛠 tv-sim](https://ews-build.webkit.org/#/builders/46/builds/4580 "Built successfully") | [✅ 🧪 mac-wk2-stress](https://ews-build.webkit.org/#/builders/8/builds/23148 "Passed tests") | | 
| [✅ 🛠 🧪 merge](https://ews-build.webkit.org/#/builders/19/builds/6311 "Built successfully and passed tests") | [✅ 🛠 watch](https://ews-build.webkit.org/#/builders/43/builds/3647 "Built successfully") | | | 
| | [✅ 🛠 watch-sim](https://ews-build.webkit.org/#/builders/45/builds/3508 "Built successfully") | | | 
<!--EWS-Status-Bubble-End-->

cdumez reviewed (changes requested):
It looks like it essentially disables the test?

It doesn't seem to wait for the load to have completed anymore before injecting the script.

cdumez reviewed (approved):
LGTM if the bots are happy.

sideshowbarker committed:
REGRESSION(269235@main?):[ Monterey+ ] http/tests/security/xss-DENIED-script-inject-into-inactive-window2-pson.html is flaky recently
https://bugs.webkit.org/show_bug.cgi?id=263879

Reviewed by Chris Dumez.

This change updates the
http/tests/security/xss-DENIED-script-inject-into-inactive-window2-pson.html
test to ensure it consistently passes.

The change adds a new isFullyActive(Document&) function to window.internals,
and uses that function in the test.

* LayoutTests/TestExpectations:
* LayoutTests/http/tests/security/xss-DENIED-script-inject-into-inactive-window2-pson-expected.txt:
* LayoutTests/http/tests/security/xss-DENIED-script-inject-into-inactive-window2-pson.html:
* Source/WebCore/dom/Document.h:
* Source/WebCore/testing/Internals.cpp:
(WebCore::Internals::isFullyActive):
* Source/WebCore/testing/Internals.h:
* Source/WebCore/testing/Internals.idl:

Canonical link: https://commits.webkit.org/270737@main
